keps/sig-auth/2579-pod-isolation-policy/README.md

```diff
@@ -0,0 +1,951 @@
+
+**Open Question:** Should this apply to all new fields, or only new fields that are constrained
+under a more recent version of the policy? If we only constrain fields that are in-scope for policy
+enforcement, then that becomes a problem if we later decide that a field should be in scope. The
+versioning machanism gives us a way to safely make "breaking" changes like adding enforcement to a
+previously unenforced field, but it is strictly less safe. Either way, there is some maintenance
+burden to declare which fields are in-scope at each version. _Leaning towards only restricting
+relevant fields._
+
+**Open Question:** If the control (or options) are strictly less privileged than the default, should
+it be allowed? For example, the `allowPrivilegeEscalation` field was added in Kubernetes v1.8, which
+sets the `no_new_privs` flag on the container processes when set to false. Prior to that version,
+the `no_new_privs` flag was never set, which is also the default in v1.8+. If older policies require
+the default value, then only `allowPrivilegeEscalation: true` would be allowed, even though
+`allowPrivilegeEscalation: false` is strictly less privileged. _Leaning towards allowing strictly
+more-restricted versions._
+
+- If the range of valid values on an existing field is expanded, the new values are forbidden. E.g.
+  adding a new enum value.
+
+Same questions as above
+
+<<[/UNRESOLVED]>>
+
+### PodTemplate Resources
+
+Audit and Warn modes are also checked on resource types that embed a PodTemplate (enumerated below),
+but Allow mode only applies to actual pod resources.
+
+Since users do not create pods directly in the typical deployment model, the warning mechanism is
+only effective if it can also warn on templated pod resources. Similarly, for audit it is useful to
+tie the audited violation back to the requesting user, so audit will also apply to templated pod
+resources. In the interest of supporting mutating admission controllers, policies will only
+be enforced on actual pods.
+
+Templated pod resources include:
+
+- v1 ReplicationController
+- v1 PodTemplate
+- apps/v1 ReplicaSet
+- apps/v1 Deployment
+- apps/v1 StatefulSet
+- apps/v1 DaemonSet
+- batch/v1 CronJob
+- batch/v1 Job
+
+CRDs that wish to take advantage of this functionality should instead have an object reference to a
+v1/PodTemplate resource. We will publish a guide (documentation and/or examples) that demonstrate
+this pattern.
+
+### Namespace policy update warnings
+
+When an allow policy (or version) label is added or changed, the admission plugin will test each pod
+in the namespace against the new policy. Violations are returned to the user as warnings. These
+checks have a timeout of XX seconds and a limit of YY pods, and will return a warning in the event
+that not every pod was checked. User exemptions are ignored by these checks, but runtime class
+exemptions still apply. Namespace exemptions are also ignored, but an additional warning will be
```

Comment:
I'm not clear on why namespace exemptions are ignored.

Reply:
The only reason you would want to set the labels on an exempt namespace would be if you wanted to remove it from the exempt list (e.g. are dry-running the new policy), or it's a mistake (in which case the warnings highlight that it's an exempt namespace).